From 2225f35b0d510a1676b256bc735eadf393d4c287 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:33 -0400
Subject: [scsi] hpsa: Add support for a few HP Storage controllers

Message-id: <1410186734-19986-43-git-send-email-thenzl@redhat.com>
Patchwork-id: 92057
O-Subject: [RHEL6.6 e-stor PATCH 42/83] hpsa: Add support for a few HP Storage controllers
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 8e616a5ee6e389f855a9fa0ab57194b4b049d9c8)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
 drivers/scsi/hpsa.c
 (we have added a group of pci-ids in a previous set)

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index cdc7fe2..f1e75d7 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -127,6 +127,11 @@ static const struct pci_device_id hpsa_pci_device_id[] = {
  {PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSI,     0x103C, 0x21CC},
  {PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSI,     0x103C, 0x21CD},
  {PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSI,     0x103C, 0x21CE},
+ {PCI_VENDOR_ID_HP_3PAR, 0x0075, 0x1590, 0x0076},
+ {PCI_VENDOR_ID_HP_3PAR, 0x0075, 0x1590, 0x0087},
+ {PCI_VENDOR_ID_HP_3PAR, 0x0075, 0x1590, 0x007D},
+ {PCI_VENDOR_ID_HP_3PAR, 0x0075, 0x1590, 0x0088},
+ {PCI_VENDOR_ID_HP, 0x333f, 0x103c, 0x333f},
  {PCI_VENDOR_ID_HP,     PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
   PCI_CLASS_STORAGE_RAID << 8, 0xffff << 8, 0},
  {0,}
@@ -182,6 +187,11 @@ static struct board_type products[] = {
  {0x21CC103C, "Smart Array", &SA5_access},
  {0x21CD103C, "Smart Array", &SA5_access},
  {0x21CE103C, "Smart Array", &SA5_access},
+ {0x00761590, "HP Storage P1224 Array Controller", &SA5_access},
+ {0x00871590, "HP Storage P1224e Array Controller", &SA5_access},
+ {0x007D1590, "HP Storage P1228 Array Controller", &SA5_access},
+ {0x00881590, "HP Storage P1228e Array Controller", &SA5_access},
+ {0x333f103c, "HP StorageWorks 1210m Array Controller", &SA5_access},
  {0xFFFF103C, "Unknown Smart Array", &SA5_access},
 };
 
-- 
1.7.1